src/Command/CodeStudio/CodeStudioPipelinesMigrateCommand.php

```diff
@@ -0,0 +1,612 @@
+    $pipelines_filepath_yaml = Path::join($this->repoRoot, 'acquia-pipelines.yaml');
+    if ($this->localMachineHelper->getFilesystem()->exists($pipelines_filepath_yml) or $this->localMachineHelper->getFilesystem()->exists($pipelines_filepath_yaml)) {
+      // print("file exists");
+      $this->gitLabClient->projects()
+        ->update($project['id'], ['ci_config_path' => '']);
+      // print_r("\nRemoved ci_config_path \n");
+      foreach (glob("acquia-pipelines.*") as $this->filename) {
+        echo "File name is : $this->filename\n";
+      }
+      $file_contents = file_get_contents($this->filename);
+      $this->acquia_pipeline_file_parse = Yaml::parse($file_contents, Yaml::PARSE_OBJECT);
+    }
+    else {
+      throw new AcquiaCliException("[Error] Missing 'acquia-pipelines.yaml' file which is required to migrate the project to Code Studio.");
+    }
+  }
+
+  /**
+   * Migrating standard template to .gitlab-ci.yml file.
+   */
+  protected function migrateCSStandardTemplate() {
+    $auto_devops_pipeline = [
+        'include' => ['project' => 'acquia/standard-template', 'file' => '/gitlab-ci/Auto-DevOps.acquia.gitlab-ci.yml'],
+      ];
+    $auto_devops_dump_file = Yaml::dump($auto_devops_pipeline);
+    $auto_devops_parse_file = Yaml::parse($auto_devops_dump_file);
+    $this->emptyArray = array_merge($this->emptyArray, $auto_devops_parse_file);
+    $this->createGitlabCiFile($this->emptyArray);
+  }
+
+  //todo : catch execpetion if any and throw error.
+
+  /**
+   * Migrating varibales to .gitlab-ci.yml file.
+   */
+  protected function migrateVariables() {
+    $varData = isset($this->acquia_pipeline_file_parse['variables'])?$this->acquia_pipeline_file_parse['variables']:FALSE;
+    if ($varData){
+      $variables_dump = Yaml::dump(['variables' => $varData]);
+      $remove_global = preg_replace('/global:/', '', $variables_dump);
+      $variables_parse = Yaml::parse($remove_global);
+      $this->emptyArray = array_merge($this->emptyArray, $variables_parse);
+      $this->createGitlabCiFile($this->emptyArray);
+      $this->io->writeln([
+        "",
+        "\n*********** Migration completed for variables section. *********\n",
+      ]);
+    }
+    else {
+      $this->io->writeln([
+        "",
+        "\n*********** acquia-pipeline file does not contain variables to migrate. *********\n",
+      ]);
+    }
+  }
+
+  /**
+   * Migrating build job to .gitlab-ci.yml file.
+   */
+  protected function migrateBuild() {
+    $buildJob=isset($this->acquia_pipeline_file_parse['events']['build']['steps'])?$this->acquia_pipeline_file_parse['events']['build']['steps']:[];
+    if (!empty($buildJob)){
+      $response_composer = NULL;
+      $response_BLT = NULL;
+
+      $chunck = array_chunk($buildJob, 1);
+      foreach ($chunck as $key => $value){
+        $keysvar = array_keys($value[0])[0];
+        if(empty($value[0][$keysvar]['script'])){
```

Comment:
You probably also need to check that this array key exists here. E.g., `array_key_exists('script', $value[0][$keysvar])`